### ISSUE TITLE
Forms: Redesign multi-select location interaction

### ISSUE THREAD

**Issue by humancompanion** [USAJOBS/design-system#68]
We need to replace the multi-select location interaction on the profile page:
https://test.usajobs.gov/Applicant/Profile/Preferences/

Noura noticed that in the Profile > Other (Applicant/Profile/Preferences/) page under question 6. Please select your desired work location(s), when the user makes some selections and saves those we display their selections in a select menu with the multiple attribute set to true. Unfortunately, webkit adds these styles to that multi-select:

select:-internal-list-box option:checked {
    background-color: -internal-inactive-list-box-selection;
    color: -internal-inactive-list-box-selection-text;
}

This in turn makes the options have a dark gray background color. That background color makes the options look at best odd, and at worst disabled. Super annoyingly the smart people behind webkit decided to deliberately not allow us web developer types to override this value:

https://bugs.chromium.org/p/chromium/issues/detail?id=398417

In short, we cannot use a multi-select dropdown for this interaction and we should come up with something custom.

**Comment by humancompanion** [comment]
We are updating this in 5.7. The new design uses an autocomplete box to find and select a location and then a list underneath with controls to delete a location. That addresses this bug. However, there are a few new wrinkles:

1. Countries need to be weighted to the top of the list (they are not currently). For example, entering "India" does not return the country of India at the top of the list making it impossible to select (the auto complete only returns 10 items and no ability to see more options). Also, the weighting should be: Country, State/Region, then City.

2. Foreign cities are not weighted as high as U.S. cities. Thus it is a bit unclear what needs to be typed in order to find, for example, "London, England". (The answer is "London, United Kingdom"). The fix here is that we are going to weight foreign cities evenly with U.S. cities and provide help text below the auto-complete field to explain what to type in order to be able to select them.

3. We have no way to "View more" beyond the initial 10 options. That's what this issue is now going to focus on.
